dask/dataframe/multi.py

```diff
@@ -927,11 +921,7 @@ def stack_partitions(dfs, divisions, join="outer"):
     for df in dfs:
         # dtypes of all dfs need to be coherent
         # refer to https://github.com/dask/dask/issues/4685
-        if is_dataframe_like(df):
```

Comment:
Why are we removing this `is_dataframe_like` check?

Reply:
Previously we did `df.columns.intersection(DataFrame)` which was just plain wrong. That always ended up in an empty `shared` so the `.astype` never occurred.

When I changed things to `df.columns.interesction(meta.columns)` to match the intent of the original code, I got failures with know & unknown categoricals.